pandas/io/formats/latex.py

```diff
@@ -256,88 +254,394 @@ def _format_multirow(
-            else:
-                label_ = f"\n\\label{{{self.label}}}"
 
-            if self.position is None:
-                position_ = ""
-            else:
-                position_ = f"[{self.position}]"
+class LatexFormatter(TableFormatter):
+    """
+    Used to render a DataFrame to a LaTeX tabular/longtable environment output.
 
-            if self.longtable:
-                table_ = f"\\begin{{longtable}}{position_}{{{column_format}}}"
-                tabular_ = "\n"
-            else:
-                table_ = f"\\begin{{table}}{position_}\n\\centering"
-                tabular_ = f"\n\\begin{{tabular}}{{{column_format}}}\n"
-
-            if self.longtable and (self.caption is not None or self.label is not None):
-                # a double-backslash is required at the end of the line
-                # as discussed here:
-                # https://tex.stackexchange.com/questions/219138
-                backlash_ = "\\\\"
-            else:
-                backlash_ = ""
-            buf.write(f"{table_}{caption_}{label_}{backlash_}{tabular_}")
-        else:
-            if self.longtable:
-                tabletype_ = "longtable"
-            else:
-                tabletype_ = "tabular"
-            buf.write(f"\\begin{{{tabletype_}}}{{{column_format}}}\n")
+    Parameters
+    ----------
+    formatter : `DataFrameFormatter`
+    column_format : str, default None
+        The columns format as specified in `LaTeX table format
+        <https://en.wikibooks.org/wiki/LaTeX/Tables>`__ e.g 'rcl' for 3 columns
 
-    def _write_tabular_end(self, buf):
+    See Also
+    --------
+    HTMLFormatter
+    """
+
+    def __init__(
+        self,
+        formatter: DataFrameFormatter,
+        longtable: bool = False,
+        column_format: Optional[str] = None,
+        multicolumn: bool = False,
+        multicolumn_format: Optional[str] = None,
+        multirow: bool = False,
+        caption: Optional[str] = None,
+        label: Optional[str] = None,
+        position: Optional[str] = None,
+    ):
+        self.fmt = formatter
+        self.frame = self.fmt.frame
+        self.longtable = longtable
+        self.column_format = column_format
+        self.multicolumn = multicolumn
+        self.multicolumn_format = multicolumn_format
+        self.multirow = multirow
+        self.caption = caption
+        self.label = label
+        self.position = position
+
+    def write_result(self, buf: IO[str]) -> None:
         """
-        Write the end of a tabular environment or nested table/tabular
-        environment.
+        Render a DataFrame to a LaTeX tabular, longtable, or table/tabular
+        environment output.
+        """
+        table_string = self.builder.product
+        buf.write(table_string)
 
-        Parameters
-        ----------
-        buf : string or file handle
-            File path or object. If not specified, the result is returned as
-            a string.
+    @property
```

Comment:
this should be a class method no?

Reply:
Not necessarily. At least not according to this https://refactoring.guru/design-patterns/builder/python/example